### PR TITLE
Add and implement Sentry

### LINKED ISSUE #9: Add Sentry error reporting
### Is your feature request related to a problem? Please describe.
It can happen that we need to know why the app is crashing for a lot of users.

### Describe the solution you'd like
We should implement error reporting to Sentry to make sure we are properly informed about the stability of the app.

### Motivation
I think I have already motivated this feature.

### Describe alternatives you've considered
There are no alternative services since we have a open source account for Sentry and concrexit already uses it as well.

### PR DESCRIPTION
**Pull request by Luca-Castelnuovo**
Fix #9

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,10 +1,21 @@
 import { createApp } from "vue";
+import * as Sentry from "@sentry/vue";
+import { Integrations } from "@sentry/tracing";
 import App from "./App.vue";
 import router from "./router";
 import store from "./store";
 import "bootstrap";
 import "bootstrap/dist/css/bootstrap.min.css";
 
+Sentry.init({
```
Comment (JobDoesburg) [inline src/main.ts:10]: Are you sure that this should be added here, and not in App.vue? (I'm not sure, the documentation says "Passing in `Vue` is optional, but if you do not pass it `window.Vue` must be present." and I'm in doubt whether `window.Vue` is present here)

File: src/main.ts [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+  dsn:
+    "https://d982883f74014ba7a7243ffaa59a3ec0@o263149.ingest.sentry.io/5757558",
```
Comment (JobDoesburg) [inline src/main.ts:12]: Can this be an (environment or repository) secret?

**Review by JobDoesburg — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 3]
```diff
@@ -9,20 +9,40 @@
...
+  tracingOptions: {
+    trackComponents: true,
+  },
+});
+
+try {
+  throw new Error("test5");
+} catch (err) {
+  Sentry.captureException(err);
+}
```
Comment (se-bastiaan) [inline src/App.vue:32]: This should obviously not end up in master :p

**Review by se-bastiaan — CHANGES_REQUESTED** [review]
I think the variables for Sentry can just be `SENTRY_ENVIRONMENT` and `SENTRY_RELEASE` next to `SENTRY_DSN` to the GitHub Actions configuration. The SENTRY_RELEASE environment can be set directly to `environment.GITHUB_SHA` and the prefix `VUE_APP` is not necessary since the Sentry package will grab those.

Unless we want to have the same prefix for all variables, in that case they have to be set on the options as well.

**Comment by Luca-Castelnuovo** [comment]
> Unless we want to have the same prefix for all variables, in that case they have to be set on the options as well.

And do we?

**Comment by se-bastiaan** [comment]
> 
> 
> > Unless we want to have the same prefix for all variables, in that case they have to be set on the options as well.
> 
> And do we?

Good point, I don't really mind having the Sentry variables separate. But I know how Sentry works. And it makes them less visible for new maintainers of this repository that have not read the Sentry docs.

So for transparency I would add them separate probably.

**Comment by Luca-Castelnuovo** [comment]
> I assume the secret should also be added in here: [`master`/.github/workflows/terraform.yml](https://github.com/svthalia/thadmin/blob/master/.github/workflows/terraform.yml)
> 
> (so add the secret in the repository settings, then edit the terraform workflow to extract the secret and use it as environment variable)

I don't have access to repo settings

**Comment by Luca-Castelnuovo** [comment]
> I think the variables for Sentry can just be `SENTRY_ENVIRONMENT` and `SENTRY_RELEASE` next to `SENTRY_DSN` to the GitHub Actions configuration. The SENTRY_RELEASE environment can be set directly to `environment.GITHUB_SHA` and the prefix `VUE_APP` is not necessary since the Sentry package will grab those.
> 
> Unless we want to have the same prefix for all variables, in that case they have to be set on the options as well.

https://cli.vuejs.org/guide/mode-and-env.html#:~:text=Note%20that%20only%20NODE_ENV%2C%20BASE_URL%2C%20and%20variables%20that%20start%20with%20VUE_APP_%20will%20be%20statically%20embedded%20into%20the%20client%20bundle%20with%20webpack.DefinePlugin.%20It%20is%20to%20avoid%20accidentally%20exposing%20a%20private%20key%20on%20the%20machine%20that%20could%20have%20the%20same%20name.

Note that only NODE_ENV, BASE_URL, and variables that start with VUE_APP_ will be statically embedded into the client bundle with webpack.DefinePlugin. It is to avoid accidentally exposing a private key on the machine that could have the same name.

**Review by se-bastiaan — APPROVED** [review]
Should have permissions now

**Comment by Luca-Castelnuovo** [comment]
Screen Shot 2021-05-12 at 12 38 05 PM

@se-bastiaan, not yet I think.

**Comment by Luca-Castelnuovo** [comment]
**TODO:** add github actions secret

**Comment by Luca-Castelnuovo** [comment]
I saw that the secret has been created.

**Review by JobDoesburg — APPROVED** [review]

**Comment by JobDoesburg** [comment]
Let's see if this works

**Pull request merged**